### PR TITLE
Remove test caching layer

### PR DESCRIPTION
**Pull request by jaredpar**
The test caching layer is inactive today and unlikely to be revived in
its current form. Test caching is much more likely to come at the Helix
layer in the future. The cache code is just making the runner harder to
maintain at this point hence removing it for now.

### CODE REVIEW

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure PTAL. Have more changes I want to make here but it's a bit simpler once I remove this old layer that we're not using anymore

**Review by jmarolf — APPROVED** [review]

**Comment by GrabYourPitchforks** [comment]
Thanks Jared! There still remains a reference to the old decommissioned Jenkins dashboard:

https://github.com/dotnet/roslyn/blob/a809acad62092c4c3eaeee997fe66dcd8a629635/src/Tools/Source/RunTests/Constants.cs#L23

**Review by chsienki — APPROVED** [review]

File: src/Tools/Source/RunTests/Program.cs [hunk 1]
```diff
@@ -348,67 +331,7 @@ private static ITestExecutor CreateTestExecutor(Options options)
...
-
-            return new CachingTestExecutor(processTestExecutor, dataStorage);
-        }
-
-        /// <summary>
-        /// Order the assembly list so that the largest assemblies come first.  This
-        /// is not ideal as the largest assembly does not necessarily take the most time.
-        /// </summary>
-        /// <param name="list"></param>
-        private static IOrderedEnumerable<string> OrderAssemblyList(IEnumerable<string> list)
```
Comment (RikkiGibson) [inline src/Tools/Source/RunTests/Program.cs:374]: was this just dead code?

**Review by RikkiGibson — APPROVED** [review]

File: src/Tools/Source/RunTests/Program.cs (hunk 1 above)
Comment (jaredpar) [reply src/Tools/Source/RunTests/Program.cs:374]: Yeah I believe an earlier PR from @jmarolf made this dead and we didn't notice

**Review by sharwell — COMMENTED** [review]
I'm totally in favor of this change provided the sequential test path remains intact 👍

**Comment by jaredpar** [comment]
> I'm totally in favor of this change provided the sequential test path remains intact 

No idea what you mean by this.

**Pull request merged**